{
  setControllerWriteExpectationToFail(true);
  final CountDownLatch failedCalledLatch=new CountDownLatch(2);
  SynStreamFrame synStreamFrame=new SynStreamFrame(SPDY.V2,SynInfo.FLAG_CLOSE,1,0,(byte)0,(short)0,null);
  IStream stream=new StandardStream(synStreamFrame.getStreamId(),synStreamFrame.getPriority(),session,null,null);
  stream.updateWindowSize(8192);
  Callback.Adapter callback=new Callback.Adapter(){
    @Override public void failed(    Throwable x){
      failedCalledLatch.countDown();
    }
  }
;
  stream.data(new StringDataInfo("data",false),5,TimeUnit.SECONDS,callback);
  stream.data(new StringDataInfo("data",false),5,TimeUnit.SECONDS,callback);
  verify(controller,times(1)).write(any(ByteBuffer.class),any(Callback.class));
  assertThat("Callback.failed has been called twice",failedCalledLatch.await(5,TimeUnit.SECONDS),is(true));
}
